xbmc/pvr/PVRPlaybackState.cpp

```diff
@@ -358,14 +359,37 @@ void CPVRPlaybackState::StartPlayback(
 
     if (item->IsPVRChannel())
     {
-      client->GetChannelStreamProperties(item->GetPVRChannelInfoTag(), props);
+      if (source == PVR_SOURCE::DEFAULT)
+      {
+        PVR_ERROR retVal = client->StreamClosed();
+        if (retVal != PVR_ERROR_NO_ERROR)
+          CLog::LogFC(LOGERROR, LOGPVR, "Client error on call to StreamClosed(): {}",
+                      CPVRClient::ToString(retVal));
+      }
+
+      client->GetChannelStreamProperties(item->GetPVRChannelInfoTag(), source, props);
+
+      if (props.LivePlaybackAsEPG())
+      {
+        const std::shared_ptr<CPVREpgInfoTag> epgTag = item->GetPVRChannelInfoTag()->GetEPGNow();
+        if (epgTag)
+        {
+          delete item;
+          item = new CFileItem(epgTag);
```

Comment:
Although it works, this is so ugly. 'item' is handed in from outside this method. If you delete it here and the calling method later on accesses the pointer - which currently is not the case but can change anytime  - this will crash.

Not sure how to solve this best. Maybe change method signature to take a `CFileItem*& item`? So the caller at least will automatically get the new item.

Reply:
Added a commit to move this use to `unique_ptr`, making it safe to change the referred to object. WDYT?

Reply:
Yep, that should do.